src/main/java/task/TaskManager.java

```diff
@@ -0,0 +1,187 @@
+        default:
+        }
+
+        Console.printNormalResponse("Got it! Added this task: ",
+                "    " + recordedTasks.get(this.recordedTasks.size() - 1).getStatusDescription(),
+                "You now have: " + this.recordedTasks.size() + " tasks");
+
+        // persist updates to local storage
+        fm.saveTasks(recordedTasks);
+    }
+
+    /**
+     * Removes a task if it exists.
+     * 
+     * @param args a list of string arguments provided by user
+     */
+    public void deleteTask(String[] args)
+            throws InsufficentArgumentsException, TaskNotFoundException, IOException {
+        if (args.length == 0) {
+            throw new InsufficentArgumentsException();
+        }
+
+        int taskNum = Integer.parseInt(args[TASK_NUMBER_INDEX]);
+
+        if (taskNum < MINIMUM_TASK_NUMBER || taskNum > recordedTasks.size()) {
+            throw new TaskNotFoundException();
+        }
+
+        String description = recordedTasks.get(taskNum - 1).getStatusDescription();
+        recordedTasks.remove(taskNum - 1);
+        Console.printNormalResponse("Noted. I've removed this task:", "    " + description,
+                "You now have: " + recordedTasks.size() + " tasks");
+        // persist updates to local storage
+        fm.saveTasks(recordedTasks);
+    }
+
+    /**
+     * Marks a task as done by calling Task.setIsDone.
+     * 
+     * @param args a list of string arguments provided by user
+     */
+    @Override
+    public final void markTask(String[] args)
+            throws InsufficentArgumentsException, TaskNotFoundException, IOException {
+        if (args.length == 0) {
+            throw new InsufficentArgumentsException();
+        }
+
+        int taskNum = Integer.parseInt(args[TASK_NUMBER_INDEX]);
+
+        if (taskNum < MINIMUM_TASK_NUMBER || taskNum > recordedTasks.size()) {
+            throw new TaskNotFoundException();
+        }
+
+        if (recordedTasks.get(taskNum - 1).getStatus()) {
+            Console.printNormalResponse("Task is already marked");
+            return;
+        }
+
+        recordedTasks.get(taskNum - 1).setStatus(true);
+        Console.printNormalResponse("I've marked this task: ",
+                recordedTasks.get(taskNum - 1).getStatusDescription());
+        // persist updates to local storage
+        fm.saveTasks(recordedTasks);
+    }
+
+    /**
+     * Marks a task as not done by calling Task.setIsDone.
+     * 
+     * @param args a list of string arguments provided by user
+     */
+    @Override
+    public final void unmarkTask(String[] args)
+            throws InsufficentArgumentsException, TaskNotFoundException, IOException {
+        if (args.length == 0) {
+            throw new InsufficentArgumentsException();
+        }
+        int taskNum = Integer.parseInt(args[TASK_NUMBER_INDEX]);
+
+        if (taskNum < MINIMUM_TASK_NUMBER || taskNum > recordedTasks.size()) {
+            throw new TaskNotFoundException();
+        }
```

Comment:
Consider refactoring repetitive code (as seen in some function above) to improve readability and reduce redundant lines of code.